### PR TITLE
Report ARIA switch controls  as "switch" with a state of on or off

### LINKED ISSUE #11310: Button with role="switch" pronounced as checkbox.
When navigating to a `<button>` with `role="switch"`, it will be pronounced as a checkbox. It is recognized as a toggle button or switch button by accessibility APIs for example, but it is not pronounced as such. 
https://www.w3.org/TR/wai-aria-1.2/#switch does state "A type of checkbox that represents on/off values", but it is not a checkbox and has a seperate role. I think it should be pronounced differently.
https://www.w3.org/TR/wai-aria-1.1/#switch does mention "Role: ROLE_SYSTEM_CHECKBUTTON" which might be the cause, but it also states "Object Attribute: xml-roles:switch" which hopefully means thiscan be fixed.

### Steps to reproduce:
1. Open https://codepen.io/erikkroes/pen/WNrOdWp in Chrome version 83.0.4103.116
2. Navigate to the example `<button>` with `role="switch"`

### Actual behavior:
Button is announced as "Checkbox".

### Expected behavior:
Button is announced as "Switch", or something alike.

### System configuration
NVDA 2020.1 installed on Parallels. Chrome version 83. Windows 10.

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
Fixes #11310

### Summary of the issue:
Currently elements on the web with an ARIA role of "switch" are presented in NVDA as checkboxes.
This is because:
* There is no switch role in IAccessible2
* ARIA practices recommend that browsers map the switch role to the toggleButton role in IAccessible2, but there is some confusion around whether the pressed or checked state should be exposed to denote the switch is on.
* NVDA historically has chosen to deliberately map switches to checkboxes with checked, rather than toggleButton with pressed.
* ARIA practices state that a switch is the same as a checkbox, just with no indeterminate state.
There is a lot of pressure from web authors and standards groups for NVDA to specifically report switch controls as having a role of "switch" rather than checkbox, as it is believed this improves communication between blind and sited individuals, as visually the switch control looks like a switch, not a checkbox.
 
### Description of user facing changes
* switch controls in Firefox, Chrome and Edge, via IAccessible2,  are now reported in NVDA as switch controls, rather than checkboxes.
	* They are spoken with a role of "switch" and state of either "on" or "off".
	* In Braille, they are represented with a role of "swtch" and a state braille pattern the same as checked / not checked.

### Description of development approach
* Added a new role of SWITCH
* Added a new state of ON
* The negative state of ON is "off".
* Only a role of switch forces a negative of the ON state. I.e. Switch is the only control that will report as "off" if no ON state exists.
* The switch role is treeted as having a text presentation type of singleLine, similar to checkbox.
* for IA2Web NVDAObjects and IAccessible2 virtualBuffers (Chrome, Firefox, Edge), IAccessible2 objects with "switch" in xml-roles:
	* map to an NVDA role of switch
	* pressed and checkable states are discarded
	* the checked state is mapped to the ON state.
The w3c ARIA examples submodule has also been updated to the latest main commit as the existing commit we were on did not yet have the switch example.

### Testing strategy:
Given a switch control such as at https://www.w3.org/WAI/ARIA/apg/example-index/switch/switch
* Open page in Firefox, Chrome and Edge
* In browse mode, use the arrow keys to move to the switch control and ensure it announces as a switch E.g. "switch on".
* In focus mode, tab to the switch control and ensure the control is reported as a switch. E.g. "Notifications switch on"
* A system test has been added which moves to the switch control in browse mode, toggles the control on and off, and reports the focus.

 
### Known issues with pull request:
For a long time I have been uncomfortable with the word "switch" as it is both a verb and a noun. Also "on" and "off" are not past tense like "checked" or "pressed". Which then means you end up with a phraise like "switch on" which could either be taken as a command (incorrectly) as in "to switch on" or as a statement (correct) "the switch is on". There is not this ambiguity with something like "checkbox checked" or "toggle button pressed". 
However, it seems that not many others are bothered by this, so I'm happy to give this ago and see where it lands with our users.

### Change log entries:
New features
Changes
Bug fixes
* ARIA switch controls are now reported as switches in Firefox, Chrome and Edge, rather than checkboxes. 
For Developers

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by codeofdusk** [comment]
> you end up with a phraise like "switch on" which could either be taken as a command (incorrectly) as in "to switch on" or as a statement (correct) "the switch is on".

Maybe consider "active" and "inactive" in place of "on" and "off"? "switch active" would be less likely considered a command.

File: tests/system/robot/chromeTests.py [hunk 1]
```diff
@@ -2104,3 +2104,125 @@ def test_focus_mode_on_focusable_read_only_lists():
 		]),
 		message="focus mode - focus list item and turn on focus mode"
 	)
+
+
+def test_ARIASwitchRole():
+	"""
+Ensure that ARIA switch controls have an appropriate role and states in browse mode.
```
Comment (seanbudd) [inline tests/system/robot/chromeTests.py:2111]: Add tab to string for consistent indentation

```suggestion
	Ensure that ARIA switch controls have an appropriate role and states in browse mode.
```

File: source/braille.py [hunk 2]
```diff
@@ -259,6 +263,8 @@
 	controlTypes.State.PRESSED: u"⢎⣀⡱",
 	# Displayed in braille when an object (e.g. a check box) is not checked.
 	controlTypes.State.CHECKED: u"⣏⣀⣹",
+	# Displayed in braille when an object (e.g. a switch control) is switched off.
+	controlTypes.State.ON: u"⣏⣀⣹",
```
Comment (seanbudd) [inline source/braille.py:267]: same here

```suggestion
	controlTypes.State.ON: "⣏⣀⣹",
```

**Review by seanbudd — COMMENTED** [review]
Looks good, only 3 minor linting issues

**Review by seanbudd — APPROVED** [review]

**Pull request merged**